server/audit_math/bravo.py

```diff
@@ -357,97 +371,78 @@ def get_sample_size(
 
     quants = [0.7, 0.8, 0.9]
 
-    samples: Dict = {}
-
     if round_sizes:
         num_sampled = sum(round_sizes.values())
         # If we've already sampled all the ballots, we should never be here
         if num_sampled >= contest.ballots:
             raise ValueError("All ballots have already been audited!")
 
-    margin = contest.margins
-    # If we're in a race without a loser, throw an error
-    if not margin["losers"]:
+    winners = contest.margins["winners"]
+    losers = contest.margins["losers"]
+
+    if not losers:
         raise ValueError("Contest must have candidates who did not win!")
 
-    # Handle landslides
-    if any([margin["winners"][winner]["p_w"] == 1.0 for winner in contest.winners]):
-        samples["asn"] = {
-            "type": "ASN",
-            "size": 1,
-            "prob": 1.0,
+    if is_tie(contest):
+        return {
+            "all-ballots": {
+                "type": "all-ballots",
+                "size": contest.ballots,
+                "prob": None,
+            }
         }
 
-        return samples
+    if is_landslide(contest):
+        return {
+            "asn": {"type": "ASN", "size": 1, "prob": 1},
+        }
 
-    # Get cumulative sample results
     cumulative_sample = {}
     if sample_results:
         cumulative_sample = compute_cumulative_sample(sample_results)
     else:
         for candidate in contest.candidates:
             cumulative_sample[candidate] = 0
 
-    asn = get_expected_sample_sizes(alpha, contest, cumulative_sample)
-
-    if asn <= 0:
-        raise ValueError("Sample indicates the audit is over!")
+    samples: Dict[str, SampleSizeOption] = {}
 
-    p_w = Decimal("inf")
-    p_l = Decimal(0)
-    best_loser = ""
-    worse_winner = ""
+    samples["asn"] = get_expected_sample_size(alpha, contest, cumulative_sample)
```

Comment:
Am I wrong that this can throw an error that isn't handled? Should it be handled here, or are you relying on something further up the call stack handling it?

Reply:
Ah, I guess technically ties and landslides are precluded by the earlier checks, so maybe we really should just crash if a ValueError gets thrown here....

Reply:
Yup exactly! My thinking was that `get_expected_sample_size` should make its limitations clear, and it's on consumers to perform relevant checks before using it